### PR TITLE
crash fix #648

### LINKED ISSUE #648: App Crashes in iPad due to transition
**I have read the [guidelines for contributing](https://github.com/jonkykong/SideMenu/blob/master/.github/CONTRIBUTING.md) and I understand**
- [yes ] My issue is happening in the **latest version** of SideMenu (older versions are no longer maintained).
- [yes ] My issue was **not** solved in the [README](https://github.com/jonkykong/SideMenu/blob/master/README.md).
- [yes] My issue can **not** be answered on [stackoverflow.com](stackoverflow.com).
- [yes ] My issue is **not** a request for new functionality that I am unwilling to build and contribute with a pull request.
- [yes] My issue **is** reproducible in the [demo project](https://github.com/jonkykong/SideMenu/tree/master/Example).

**Describe the bug**
When app goes to background/when we try to open an external link and then when we come back to our app, there is a crash in SideMenuTransitionController (at presentationController.dismissalTransitionWillBegin() of transitionWillBegin) in iPad.

**To Reproduce**

Create an option under Push View Controller of Left Menu that takes to an external link.
Click on the above option.
Come back to Side Menu app and click on Left Menu.

**Expected behavior**
Side Menu does not crash in iPhone and behaves as expected. However, crash occurs in iPad.

**Video**
[CrashRecording.zip](https://github.com/jonkykong/SideMenu/files/5266252/CrashRecording.zip)

**Additional context**
Please find two different crash logs from the actual app in which I used the Side Menu :

1.
Crashed: com.apple.main-thread
EXC_BREAKPOINT 0x000000010172cf4c
0
SideMenu
SideMenuTransitionController.swift - Line 104
closure #1 in SideMenuNavigationController.viewWillTransition(to:with:) + 104
1
SideMenu
- Line 4323856260
thunk for @escaping @callee_guaranteed (@guaranteed UIViewControllerTransitionCoordinatorContext) -> () + 4323856260
2
UIKitCore
-[_UIViewControllerTransitionCoordinator _applyBlocks:releaseBlocks:] + 264
3
UIKitCore
-[_UIViewControllerTransitionContext __runAlongsideAnimations] + 256
4
UIKitCore
-[_UIWindowAnimationController animateTransition:] + 328
5
UIKitCore
-[UIWindow _adjustSizeClassesAndResizeWindowToFrame:] + 932
6
UIKitCore
-[UIWindow _resizeWindowFrameToSceneBoundsIfNecessary] + 248
7
UIKitCore
__78-[UIWindow _rotateWindowToOrientation:updateStatusBar:duration:skipCallbacks:]_block_invoke + 388
8
UIKitCore
__58-[_UIWindowRotationAnimationController animateTransition:]_block_invoke_2 + 172
9
UIKitCore
+[UIView(Internal) _performBlockDelayingTriggeringResponderEvents:forScene:] + 204
10
UIKitCore
__58-[_UIWindowRotationAnimationController animateTransition:]_block_invoke + 164
11
UIKitCore
-[_UIWindowRotationAnimationController animateTransition:] + 492
12
UIKitCore
-[UIWindow _rotateToBounds:withAnimator:transitionContext:] + 580
13
UIKitCore
-[UIWindow _rotateWindowToOrientation:updateStatusBar:duration:skipCallbacks:] + 1420
14
UIKitCore
-[UIWindow _setRotatableClient:toOrientation:updateStatusBar:duration:force:isRotating:] + 408
15
UIKitCore
-[UIWindow _setRotatableViewOrientation:updateStatusBar:duration:force:] + 128
16
UIKitCore
-[_UIFullscreenPresentationController _placeCounterRotationViewWithView:inWindow:fromOrientation:toOrientation:force:] + 504
17
UIKitCore
-[_UIFullscreenPresentationController _adjustOrientationIfNecessaryInWindow:forViewController:preservingViewController:] + 880
18
UIKitCore
-[UIPresentationController _presentWithAnimationController:interactionController:target:didEndSelector:] + 932
19
UIKitCore
-[UIViewController _presentViewController:modalSourceViewController:presentationController:animationController:interactionController:completion:] + 1412
20
UIKitCore
-[UIViewController _presentViewController:withAnimationController:completion:] + 4228
21
UIKitCore
__63-[UIViewController _presentViewController:animated:completion:]_block_invoke + 104
22
UIKitCore
-[UIViewController _performCoordinatedPresentOrDismiss:animated:] + 508
23
UIKitCore
-[UIViewController _presentViewController:animated:completion:] + 196
24
UIKitCore
-[UIViewController presentViewController:animated:completion:] + 164

2.
Crashed: com.apple.main-thread
EXC_BREAKPOINT 0x0000000101944f4c
0
SideMenu
(Missing)
1
SideMenu
(Missing)
2
UIKitCore
-[_UIViewControllerTransitionCoordinator _applyBlocks:releaseBlocks:] + 264
3
UIKitCore
-[_UIViewControllerTransitionContext __runAlongsideAnimations] + 256
4
UIKitCore
-[_UIWindowAnimationController animateTransition:] + 328
5
UIKitCore
-[UIWindow _adjustSizeClassesAndResizeWindowToFrame:] + 932
6
UIKitCore
-[UIWindow _resizeWindowFrameToSceneBoundsIfNecessary] + 248
7
UIKitCore
__78-[UIWindow _rotateWindowToOrientation:updateStatusBar:duration:skipCallbacks:]_block_invoke + 388
8
UIKitCore
__58-[_UIWindowRotationAnimationController animateTransition:]_block_invoke_2 + 172
9
UIKitCore
+[UIView(Internal) _performBlockDelayingTriggeringResponderEvents:forScene:] + 204
10
UIKitCore
__58-[_UIWindowRotationAnimationController animateTransition:]_block_invoke + 164
11
UIKitCore
-[_UIWindowRotationAnimationController animateTransition:] + 492
12
UIKitCore
-[UIWindow _rotateToBounds:withAnimator:transitionContext:] + 580
13
UIKitCore
-[UIWindow _rotateWindowToOrientation:updateStatusBar:duration:skipCallbacks:] + 1420
14
UIKitCore
-[UIWindow _setRotatableClient:toOrientation:updateStatusBar:duration:force:isRotating:] + 408
15
UIKitCore
-[UIWindow _setRotatableViewOrientation:updateStatusBar:duration:force:] + 128
16
UIKitCore
-[_UIFullscreenPresentationController _placeCounterRotationViewWithView:inWindow:fromOrientation:toOrientation:force:] + 504
17
UIKitCore
-[_UIFullscreenPresentationController _adjustOrientationIfNecessaryInWindow:forViewController:preservingViewController:] + 880
18
UIKitCore
-[UIPresentationController _presentWithAnimationController:interactionController:target:didEndSelector:] + 932
19
UIKitCore
-[UIViewController _presentViewController:modalSourceViewController:presentationController:animationController:interactionController:completion:] + 1412
20
UIKitCore
-[UIViewController _presentViewController:withAnimationController:completion:] + 4228
21
UIKitCore
__63-[UIViewController _presentViewController:animated:completion:]_block_invoke + 104
22
UIKitCore
-[UIViewController _performCoordinatedPresentOrDismiss:animated:] + 508
23
UIKitCore
-[UIViewController _presentViewController:animated:completion:] + 196
24
UIKitCore
-[UIViewController presentViewController:animated:completion:] + 164

In case of any clarification, please feel free to reach out to me.
Thanks.

Here's the minimal modified version of demo(example[SideMenu-master.zip])
[SideMenu-master.zip](https://github.com/jonkykong/SideMenu/files/5266260/SideMenu-master.zip)

### PR DESCRIPTION
**Pull request by 2h4u**
Not to a 100% sure if this a good approach for the problem but it fixes the crash i experienced with no discoverable side-effects.

### CODE REVIEW

**Comment by jonkykong** [comment]
Thank you for your contribution, @2h4u.

In reviewing crashes related to this issue (https://github.com/jonkykong/SideMenu/issues/648), it appears that https://github.com/jonkykong/SideMenu/pull/642 will also solve the crashes occurring on iPad. I'm more inclined to accept this fix as more robust unless you have any objections.

**Comment by 2h4u** [comment]
My only objection is that this was the first time that I took a closer look at the code of this project and therefore I am not fully familiar with the code base. At this situation I always struggle to be a 100% confident 😅
But to this point I did not experience any issues with this fix.

**Comment by jonkykong** [comment]
No worries. I appreciate you taking the time!

File: Pod/Classes/SideMenuNavigationController.swift [hunk 1]
```diff
@@ -264,8 +264,9 @@ open class SideMenuNavigationController: UINavigationController {
     override open func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
         super.viewWillTransition(to: size, with: coordinator)
         
-        // Don't bother resizing if the view isn't visible
-        guard let transitionController = transitionController, !view.isHidden else { return }
+      // Don't bother resizing if the view isn't visible or in background
+      guard let transitionController = transitionController, !view.isHidden,
+            UIApplication.shared.applicationState != UIApplication.State.background else { return }
```
Comment (DiegoArenaO) [inline Pod/Classes/SideMenuNavigationController.swift:269]: This change fix the issue only if the device is not rotated, but it's a good approach to fix the rotation issue. 
Thanks

Comment (DiegoArenaO) [reply Pod/Classes/SideMenuNavigationController.swift:269]: Adding the inactive case to your fix I was able to fix the weird transition when rotates the device in background too.

guard let transitionController = transitionController, !view.isHidden, 
           UIApplication.shared.applicationState != UIApplication.State.background, 
           UIApplication.shared.applicationState != UIApplication.State.inactive else { return }

**Pull request closed without merging**